src/Bootstrap/Form/Range.elm

```diff
@@ -0,0 +1,219 @@
+module Bootstrap.Form.Range exposing
+    ( range, Range
+    , id, min, max, step, disabled, onInput, attrs, Option
+    , success, danger
+    )
+
+{-| This module allows you to create Bootstrap styled range inputs.
+
+
+# Creating
+
+@docs range, Range
+
+
+# Options
+
+@docs id, min, max, step, disabled, onInput, disabled, attrs, Option
+
+
+# Validation
+
+@docs success, danger
+
+-}
+
+import Bootstrap.Form.FormInternal as FormInternal
+import Html
+import Html.Attributes as Attributes
+import Html.Events as Events
+
+
+{-| Opaque composable type representing a Radio.
+-}
+type Range msg
+    = Range
+        { options : List (Option msg)
+        }
+
+
+{-| Opaque type representing valid customization options for a radio
+-}
+type Option msg
+    = Id String
+    | Min String
+    | Max String
+    | Step String
+    | OnInput (Maybe (String -> msg))
+    | Disabled Bool
+    | Validation FormInternal.Validation
+    | Attrs (List (Html.Attribute msg))
+
+
+type alias Options msg =
+    { id : Maybe String
+    , min : Maybe String
+    , max : Maybe String
+    , step : Maybe String
+    , disabled : Bool
+    , onInput : Maybe (String -> msg)
+    , validation : Maybe FormInternal.Validation
+    , attributes : List (Html.Attribute msg)
+    }
+
+
+{-| Create a range input.
+
+    Range.range
+        [ Range.id "myRange"
+        , Range.min -10
+        , Range.max 10
+        , Radio.onInput MyRangeMsg
+        ]
+
+-}
+range : List (Option msg) -> Html.Html msg
+range options =
+    create options |> view
+
+
+create : List (Option msg) -> Range msg
+create options =
+    Range
+        { options = options
+        }
+
+
+addOption : Option msg -> Range msg -> Range msg
+addOption opt (Range ({ options } as range_)) =
+    Range { range_ | options = opt :: options }
+
+
+view : Range msg -> Html.Html msg
+view (Range range_) =
+    let
+        opts =
+            List.foldl applyModifier defaultOptions range_.options
+    in
+    Html.input (toAttributes opts) []
+
+
+{-| Shorthand for assigning an onInput handler for a range.
+-}
+onInput : Maybe (String -> msg) -> Option msg
+onInput toMsg =
+    OnInput toMsg
+
+
+{-| Option to set the min value for the range.
+-}
+min : String -> Option msg
+min val =
+    Min val
+
+
+{-| Option to set the max value for the range.
+-}
+max : String -> Option msg
+max val =
+    Max val
+
+
+{-| Option to set the step value for the range.
+-}
+step : String -> Option msg
```

Comment:
Could be number ? If so, we should add an `anyStep` function as well (ref https://getbootstrap.com/docs/4.3/components/forms/#range-inputs)

Reply:
I'm not sure I understand what the anyStep function should do? 

With regards to `step`, `min` and `max` taking `number` as input, I don't think that's possible as there's no `number -> String` function as far as I'm aware. We could use floats, though. Would you prefer that?

Reply:
from mdn "A string value of any means that no stepping is implied, and any value is allowed (barring other constraints, such as min and max)."

number -> string: Nah I think it's probably best to just stick with strings tbh.

Reply:
> from mdn "A string value of any means that no stepping is implied, and any value is allowed (barring other constraints, such as min and max)."

Good point! I assumed that was the default behavior, but I see that we'll need an anyStep.

Reply:
Although, perhaps now that we're making `step` take a `String` as input, perhaps it's not needed anyway?

Reply:
Probably not worth it, as we accept strings we pretty much give up on type safety anyway.

Reply:
In that case I think I've resolved all the issues you pointed out.